automated-comments/go/hamming/comparing_bytes_disguised_as_runes.md

```diff
@@ -0,0 +1,4 @@
+Converting the `bytes` to `runes` to compare disguises the problem.
+It works perfectly in the context of this exercise as there are only ascii letters in DNA strands.
+
+Try creating more tests with non-ascii characters and see what happens.
```

Comment:
I think this is likely to be a bit beyond the capabilities of students so early in the track, and unless they do that, they won't get the point. Do we need a more detailed explanation of the problem here?

I'm not completely clear what triggers this comment, but I'm guessing it's something like indexing the string by byte, and then converting that byte to a rune before comparing it. The right thing to do (I usually propose) is to convert both strings to rune sequences _first_, before doing anything with them. Then it's meaningful to compare their lengths, and safe to iterate over them comparing element with element.

Reply:
```suggestion
Try creating more tests with non-ASCII characters and see what happens.
```

(same fix as above).

For the rest I agree with @bitfield .

Reply:
Yes, this is triggered when doing something like this:

```go
for i, r := range a {
    if r == rune(b[i])
}
```

I'll look into explaining the "problem" better. I think I would stick with letting the user find out what to do about it as this is more of a challenge rather than a "must". That should probably be clearer as well.